client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/index.tsx

```diff
@@ -90,6 +90,14 @@ const SenseiPlan: Step = ( { flow, navigation: { submit } } ) => {
 		[ businessPlan.productSlug, domain, flow, senseiProPlan.productSlug ]
 	);
 
+	const isLoading = ! businessPlan.monthlyPrice || ! senseiProPlan.monthlyPrice;
+
+	const onBillingPeriodChange = ( value: Plans.PlanBillingPeriod ) => {
+		if ( ! isLoading ) {
```

Comment:
> I updated it to prevent changing the billing period while it's loading. https://github.com/Automattic/wp-calypso/commit/aaf3ac5dd038df5e4e877d9f202a1b478db111e1

Thanks for trying to mitigate it @renatho!

The issue I'm seeing now is that the tab stays unresponsive for a while, longer if the speed is slow. I'm not sure which issue outweighs which, a short appearance of the NaN option in the tab, or the tab switcher being unresponsive for a while during which time user may click on it many times but not see anything happening but after a while, clicking on it suddenly works.

For me personally, suddenly seeing a wrong value felt less of an issue between these two. As we don't have any visual way to know here that the tab buttons are disabled, clicking on them and not seeing anything happening bothered me more, because clicking was an action that I actively took as a user and nothing happened whereas the nan option a passive thing that I just happened to notice but it didn't stop me from doing anything else.

Do you think we can take any other approach? For example -
- Handling it to not show 'NaN'
- Or, show something visually perceivable to make user not click it immediately, like a disabled state view, or a loader

Or maybe, we can open a low priority issue to handle it, as this is not too problematic, revert the changes we've made here, and leave it to be handled as part of that and merge the tooltip related changes, as they're more pressing. LMKWYT! ☺️

Reply:
Thanks for adding the loader @renatho! It's working as expected!

> I implemented a loader similar to the one we have in the other parts.

A thought just occurred to me while checking the other parts we have a loader on and I wanted to share with you. As you can see in the screenshot above, the loader is only on the part of the data that's prone to frequent updates, which is the price, not the rest, like the bullet points. So my point is, as the tab switcher is not actually prone to frequent updates or the options themselves are not fetched from the backend, should there be a loader on the switcher.

I'm okay with merging it as it is now, just wanted to share my thoughts to know your thoughts about this! :) Sorry for so many to and fros!

Reply:
Hi @Imran92!

No need to sorry! It was very helpful!
I think now we have a much better and simple solution. https://github.com/Automattic/wp-calypso/pull/84175/commits/023496f541523ab6dd8bf6eb5f247ba427994ce8

Basically, it will only display the tooltip after the prices are loaded. So the `%` discount won't change while loading, no NaN, and no loader.

Let me know what you think, so if it's good we can deploy it.

Reply:
Nice! I like this solution 👍 Simple and isn't blocking anything. LGTM! 🚀 Thanks @renatho